packages/yarnpkg-builder/sources/commands/build/plugin.ts

```diff
@@ -17,11 +17,13 @@ const pathRegExp = /^(?![a-zA-Z]:[\\/]|\\\\|\.{0,2}(?:\/|$))((?:@[^/]+\/)?[^/]+)
 // their own (@arcanis/yarn-plugin-foo would override @yarnpkg/plugin-foo
 // as well as @mael/yarn-plugin-foo)
 const getNormalizedName = (name: string) => {
-  const parsing = name.match(/^(?:@yarnpkg\/|(?:@[^/]+\/)?yarn-)(plugin-[^/]+)/);
+  const parsing = name.match(/^(?:(@yarnpkg\/)|(@[^/]+\/)?yarn-)(plugin-[^/]+)$/);
```

Comment:
This could use `structUtils.tryParseIdent` then validate the `name` directly instead of this regex

Reply:
Not sure but I just realized use regexp to test and just return name is good enough for this purpose, what do you think about?

Reply:
Just something like
```typescript
const getNormalizedName = (name: string) => {
  if (!/^(?:@yarnpkg\/|(?:@[^/]+\/)?yarn-)(plugin-[^/]+)/.test(name))
    throw new UsageError(`Invalid plugin name "${name}" - it should be "yarn-plugin-<something>"`);

  return name;
};
```